File: golang/10_canary_sdk/contain.go
package _0_canary_sdk

// Contains checks whether a given element is present in the slice.
func Contains[T comparable](slice []T, value T) bool {
	for _, v := range slice {
		if v == value {
			return true
		}
	}
	return false
}

// ContainsAny checks whether any of the given elements are present in the slice.
func ContainsAny[T comparable](slice []T, values ...T) bool {
	for _, val := range values {
		if Contains(slice, val) {
			return true
		}
	}
	return false
}

// ContainsAll checks whether all of the given elements are present in the slice.
func ContainsAll[T comparable](slice []T, values ...T) bool {
	for _, val := range values {
		if !Contains(slice, val) {
			return false
		}
	}
	return true
}

File: myjava/src/main/java/org/example/canary/Contain.java
package org.example.canary;

import java.util.List;

public class Contain {
    // Contains checks whether a given element is present in the list.
    public static boolean contains(List<String> list, String value) {
        return list.contains(value);
    }

    // ContainsAny checks whether any of the given elements are present in the list.
    public static boolean containsAny(List<String> list, String... values) {
        for (String val : values) {
            if (list.contains(val)) {
                return true;
            }
        }
        return false;
    }

    // ContainsAll checks whether all of the given elements are present in the list.
    public static boolean containsAll(List<String> list, String... values) {
        for (String val : values) {
            if (!list.contains(val)) {
                return false;
            }
        }
        return true;
    }
}
